File: weather/src/main/java/com/github/xiaogegechen/weather/adapter/SettingAdapter.java
package com.github.xiaogegechen.weather.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Switch;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.github.xiaogegechen.weather.R;
import com.github.xiaogegechen.weather.model.Setting;

import java.util.List;

/**
 * 设置界面的listView的adapter
 */
public class SettingAdapter extends ArrayAdapter<Setting> {

    private int mResource;
    private  List<Setting> mList;

    public SettingAdapter(@NonNull Context context, int resource, @NonNull List<Setting> objects) {
        super(context, resource, objects);
        mResource = resource;
        mList = objects;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View view;
        ViewHolder viewHolder;
        if(convertView == null){
            view = LayoutInflater.from(getContext()).inflate(mResource, parent, false);
            viewHolder = new ViewHolder(view);
            view.setTag(viewHolder);
        }else{
            view = convertView;
            viewHolder = (ViewHolder) view.getTag();
        }
        Setting setting = mList.get(position);
        viewHolder.mTextView.setText(setting.getName());
        if(setting.isSwitchVisibility()){
            viewHolder.mSwitch.setVisibility(View.VISIBLE);
            viewHolder.mSwitch.setChecked(setting.isSwitchChecked());
        }else{
            viewHolder.mSwitch.setVisibility(View.GONE);
        }
        return view;
    }

    static class ViewHolder{
        Switch mSwitch;
        TextView mTextView;

        ViewHolder(View itemView){
            mSwitch = itemView.findViewById(R.id.switchButton);
            mTextView = itemView.findViewById(R.id.title);
        }
    }
}

File: module_left/src/main/java/com/github/xiaogegechen/module_left/adapter/SettingAdapter.kt
package com.github.xiaogegechen.module_left.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.Switch
import android.widget.TextView
import com.github.xiaogegechen.module_left.R
import com.github.xiaogegechen.module_left.model.Setting

class SettingAdapter(var mContext: Context, var resource: Int, var settingList: MutableList<Setting>): ArrayAdapter<Setting>(mContext, resource, settingList) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val setting = getItem(position)
        val view: View
        val holder: ViewHolder
        if(convertView == null){
            view = LayoutInflater.from(mContext).inflate(resource, parent, false)
            holder = ViewHolder(view)
            view.tag = holder
        }else{
            view = convertView
            holder = view.tag as ViewHolder
        }

        holder.mTitle?.text = setting?.name
        if(setting!!.switchVisibility){
            holder.mSwitch?.visibility = View.VISIBLE
        }else{
            holder.mSwitch?.visibility = View.INVISIBLE
        }

        return view
    }

    private class ViewHolder(view: View?){
        var mSwitch: Switch? = view?.findViewById(R.id.setSwitch)
        var mTitle: TextView? = view?.findViewById(R.id.setTitle)
    }
}